engine/kv_engine_hash.cpp

```diff
@@ -0,0 +1,242 @@
+#include "kv_engine.hpp"
+
+namespace KVDK_NAMESPACE {
+Status KVEngine::HashLength(StringView key, size_t* len) {
+  if (!CheckKeySize(key)) {
+    return Status::InvalidDataSize;
+  }
+  HashList* hlist;
+  Status s = hashListFind(key, &hlist, false);
+  if (s != Status::Ok) {
```

Comment:
if (s==Status::OutDated) return Status::NotFound

Reply:
It's up to glue layer to check this. 
We should discuss this again.

Reply:
But we discuss before: outDated is internal status.

Reply:
reference: sorted, string